### ISSUE TITLE
Excalidraw: Scrolling causes some weirdness

### ISSUE THREAD

**Issue by jasonLaster** [RecordReplay/devtools#2157]
**Description:**

* Clicking on the log messages in the console does not always update the video
* Clicking next on the breakpoint hits sometimes jumps several hits

**Steps to Reproduce:**

1. go to [excalidraw](https://replay.io/view?id=3a3c7c94-7dca-4a6b-82fa-a12a776006f0) 
2. Add a breakpoint in App.tsx:974 a log expression `"render", this.state.scrollX` 
3. Notice how the scroll position changes, but the video only changes once every ten logs

**Expected Result:**

**Additional Context:**

* Replay:
https://replay.io/view?id=aded64ab-4cab-4ae8-893f-42b087a7b4e3

**Comment by jasonLaster** [comment]
The repro for the bug is

1. go to excalidraw.com
2. If you load a shared scene with at least another person on the scene
3. you can start seeing the content via the firebase response. 
4. If you scroll and you receive the response from the websocket INIT, then it scrolls you back to the center which is jarring.

**Comment by jasonLaster** [comment]
I left a couple of comments in the replay, but was surprised that the 168 hits came back out of order. 

What's especially weird about this is i wonder if the hits and console logs are in the same order because the logs look right as in the scrollX seems to be in sequential order..


<img width="1329" alt="Screen Shot 2021-03-20 at 6 35 38 PM" src="https://user-images.githubusercontent.com/254562/111890902-22220580-89ab-11eb-8dee-e5552dd9a866.png">

**Comment by jasonLaster** [comment]
this is better but still not perfect. If you log  "componentDidUpdate", this.state.scrollX, you can see that it changes on every render but you don't get a new screenshot on every render

we only call componentDidUpdate when something changes, so there should be a screenshot for every single update

**Comment by bhackett1024** [comment]
> this is better but still not perfect. If you log "componentDidUpdate", this.state.scrollX, you can see that it changes on every render but you don't get a new screenshot on every render
> 
> we only call componentDidUpdate when something changes, so there should be a screenshot for every single update

What seems most likely here is that the browser is not painting after every componentDidUpdate call, and since we only show the graphics at the last paint the screen shot will frequently be out of sync with the DOM if there are many of these updates.  It's hard to force the browser to paint when we want it to; the browser provides a requestAnimationFrame API for syncing up JS changes with repaints, but I think this feature is pretty specialized for things like animations (!) and games which need precise control over how repaints happen.

So things are behaving as expected with how things are currently designed, but it would be much nicer if we redesigned this so that we could paint the current DOM.  That's a new feature, not a bugfix; filed https://github.com/RecordReplay/devtools/issues/2166.

**Comment by loganfsmyth** [comment]
> I left a couple of comments in the replay, but was surprised that the 168 hits came back out of order.
> 
> What's especially weird about this is i wonder if the hits and console logs are in the same order because the logs look right as in the scrollX seems to be in sequential order..

This is a bit hard to follow, but I _think_ https://github.com/RecordReplay/devtools/pull/2167 to address this ordering weirdness.

**Comment by hbenl** [comment]
> What's especially weird about this is i wonder if the hits and console logs are in the same order because the logs look right as in the scrollX seems to be in sequential order..

No, they're not in the same order, console messages are always sorted by ExecutionPoint.

**Comment by hbenl** [comment]
> This is a bit hard to follow, but I think #2167 to address this ordering weirdness.

Yes, it does, as does #2164 (which is almost identical).
